### PR TITLE
Revit_Engine: Modify and Query descriptions added and GetProperty/SetProperty duplicates deprecated

### LINKED ISSUE #595: Add and update descriptions to Revit_oM and Revit_Engine
#### What is missing/incorrect?

As in the title - many descriptions are missing, some of the existing are wrong or obsolete. A general cleanup needed.

### LINKED ISSUE #617: Revit_Toolkit: deprecate SetProperty and GetProperty duplicate methods in Revit_Engine
#### Description:

A few methods in Revit_Engine do literally nothing beyond setting the property, which can (and should?) be achieved with `SetProperty` or `GetProperty`. To be deprecated.

### LINKED ISSUE #618: Revit_Toolkit: BH.Engine.Adapters.Revit.Query.Duplicate is a duplicate of BH.Engine.Adapters.Revit.Modify.RemoveIdentifiers
#### Description:
As in the title, exactly the same.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #595
Closes #617
Closes #618

### Test files

Not needed - the changes should be non-breaking.

### Changelog

- `BH.Engine.Adapters.Revit.Modify` and `BH.Engine.Adapters.Revit.Query` descriptions added and/or refreshed
- logical duplicates of `GetProperty` and `SetProperty` (i.e. the methods that lead to results achievable with the latter in a single step) have been deprecated
- `BH.Engine.Adapters.Revit.Query.Duplicate` method has been deprecated

### Additional comments

There should be no conflicts with #616, so these can be reviewed independently.

### CODE REVIEW

File: Revit_Engine/Modify/AddMap.cs [hunk 1]
```diff
@@ -53,12 +52,12 @@ public static TypeMap AddMap(this TypeMap typeMap, string sourceName, string des
...
-        [Input("sourceName", "BHoM parameter name of type")]
-        [Input("destinationNames", "Revit parameter names to be mapped")]
-        [Output("TypeMap")]
+        
+        [Description("Links Revit parameters with BHoM type property or CustomData inside existing TypeMap.")]
+        [Input("typeMap", "TypeMap to be extended.")]
+        [Input("sourceName", "BHoM type property or CustomData name.")]
+        [Input("destinationName", "Revit parameter name to be mapped.")]
+        [Output("typeMap")]
         public static TypeMap AddMap(this TypeMap typeMap, string sourceName, IEnumerable<string> destinationNames)
```
Comment (kThorsager) [inline Revit_Engine/Modify/AddMap.cs]: These don't match

Comment (kThorsager) [reply Revit_Engine/Modify/AddMap.cs]: Can't believe I found the only compliance error

Comment (kThorsager) [reply Revit_Engine/Modify/AddMap.cs]: But regardless, would not mind some explanation on how one maps from one BHoM param name to many Revit ones

File: Revit_Engine/Query/InnerPolyCurves.cs [hunk 2]
```diff
@@ -38,10 +34,10 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
-        [Description("Returns the inner PolyCurves from list. QUICK CHECK (BOUNDING BOX)")]
-        [Input("polyCurve", "PolyCurve")]
-        [Input("polyCurves", "PolyCurves")]
-        [Output("PolyCurves")]
+        [Description("Returns the PolyCurves from list that lie inside given outer PolyCurve. QUICK CHECK (BOUNDING BOX).")]
```
Comment (kThorsager) [inline Revit_Engine/Query/InnerPolyCurves.cs]:
```suggestion
        [Description("Returns the PolyCurves from the list 'polyCurves' that lie inside the provided PolyCurve. QUICK CHECK (BOUNDING BOX).")]
```

File: Revit_Engine/Query/Names.cs [hunk 3]
```diff
@@ -34,6 +35,11 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Returns a collection of parameter names associated with a given type property inside MapSettings.")]
+        [Input("mapSettings", "MapSettings to be queried.")]
+        [Input("type", "Type to be sought for.")]
+        [Input("name", "Property name to be sought for.")]
+        [Output("names")]
         public static IEnumerable<string> Names(this MapSettings mapSettings, Type type, string name)
```
Comment (kThorsager) [inline Revit_Engine/Query/Names.cs]: Is the input `name` what the output `names` is? Don't think I get this one

File: Revit_Engine/Query/InnerPolyCurves.cs (hunk 2 above)
Comment (pawelbaran) [reply Revit_Engine/Query/InnerPolyCurves.cs]: I have updated the description, but would be worried about that tbh... This method is legacy, first one to be deprecated in the next Milestone (1. it should sit in geometry 2. it is a duplicate of geometry 3. it is case-specific).

File: Revit_Engine/Query/Names.cs (hunk 3 above)
Comment (pawelbaran) [reply Revit_Engine/Query/Names.cs]: You look for Revit parameter names that are linked with a certain type property name, I agree that this does not look good, but it is legacy to be revisited in the coming Milestone - I would be comfortable with leaving it as is for now and giving it a deeper thought in a month from now.

File: Revit_Engine/Query/CustomDataValue.cs [hunk 4]
```diff
@@ -33,10 +32,10 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
-        [Description("Gets value of CustomData for given name.")]
-        [Input("bHoMObject", "BHoMObject")]
-        [Input("name", "CustomData Name")]
-        [Output("Value")]
+        [Description("Gets value of BHoM objects's CustomData under given key.")]
+        [Input("bHoMObject", "BHoMObject to be queried.")]
+        [Input("name", "CustomData key to be sought for.")]
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Revit_Engine/Query/CustomDataValue.cs]: Agreed, I did not like mixing _name_ with _key_ but did not have balls to change it so late before the beta. Tested it now though, change from `name` to `key` is picked fine by Versioning so `key` it is.

Coming to your question, to large extent it is a duplicate of `GetProperty`, but it does a null check on `CustomData` etc., so helps avoiding boilerplate - that is why I decided not to deprecate it now. Instead, I have a todo item for the next Milestone to merge it into BHoM_Engine as it definitely is too generic for Revit_Toolkit. @IsakNaslundBh we had a chat about it some time ago, please shout out if I messed things up.

**Pull request merged**